pkg/security/security_profile/profile/profile_dir.go

```diff
@@ -271,6 +275,30 @@ func (dp *DirectoryProvider) getProfiles() map[cgroupModel.WorkloadSelector]prof
 	return dp.profileMapping
 }
 
+// OnLocalStorageCleanup removes the provided files from the entries of the directory provider
+func (dp *DirectoryProvider) OnLocalStorageCleanup(files []string) {
+	dp.Lock()
+	defer dp.Unlock()
+
+	fileMask := make(map[string]bool)
+	for _, file := range files {
+		if path.Ext(file) == profileExtension {
+			fileMask[file] = true
+		}
+	}
+
+	// iterate through the list of profiles and remove the entries that are about to be deleted from the file system
+	for selector, fsEntry := range dp.profileMapping {
+		if _, found := fileMask[fsEntry.path]; found {
+			delete(dp.profileMapping, selector)
```

Comment:
Is it safe to iterate on a slice where we can evict cell when iterating ?

Reply:
Yes 🎉  https://stackoverflow.com/questions/23229975/is-it-safe-to-remove-selected-keys-from-map-within-a-range-loop